File: ScopesClosures/add_nums.js
function outerFunction(x) {
    let message = `Outer Funnction! x = ${x}`;
    console.log(message)
    function innerFunction(y) {
        console.log(message);
        console.log(`Sum of x and y: ${x + y}`); // can access x from outer function
    }

    return innerFunction;
}

const closure = outerFunction(5);
closure(10);

File: ScopesClosures/add_nums.py
def outer_function(x):
    message = f"Outer Funnction! x = {x}"
    print(message)
    def inner_function(y):
        print(message)
        print(f"Sum of x and y: {x + y}")  # can access x from outer function

    return inner_function

closure = outer_function(5)
closure(10)
